src/middlewared/middlewared/plugins/snapshot.py

```diff
@@ -377,6 +377,17 @@ def max_count(self):
         # with too many, then File Explorer will show no snapshots available.
         return 512
 
+    @accepts()
+    @returns(Int())
+    def max_total_count(self):
+        """
+        Returns a maximum amount of snapshots (total) the system can sustain.
+        """
+
+        # Having too many snapshots results in various performance complications (mainly, when listing them).
+        # This is a random round number that is large enough and does not cause issues in most use cases.
+        return 10000
```

Comment:
should this be a hard coded value like this ?

Reply:
@rick-mesta what are other options?

Reply:
A tuneable that doesn't require us to rebuild the software, perhaps ?

I'm just thinking that what works for one customer might not necessarily work for another. So this might be a way to provide relief in case 10k is a low ceiling for some. I don't know how these types of tuneables are available in TrueNAS.

Reply:
@william-gr what's our current position wrt tuneables like @rick-mesta proposes?